File: Fortran/0154/0154_0016.f90
module m1
  external ATOMIC_DEFINE,ATOMIC_REF,CO_SUM,CO_MIN,CO_MAX
  external IMAGE_INDEX,LCOBOUND,NUM_IMAGES,THIS_IMAGE,UCOBOUND
end
use m1
k=0
if (k==1) then
call sub( &
   ATOMIC_DEFINE,ATOMIC_REF,CO_SUM,CO_MIN,CO_MAX, &
   IMAGE_INDEX,LCOBOUND,NUM_IMAGES,THIS_IMAGE,UCOBOUND)
endif
print *,'pass'
end
subroutine  sub( &
   ATOMIC_DEFINE,ATOMIC_REF,CO_SUM,CO_MIN,CO_MAX, &
   IMAGE_INDEX,LCOBOUND,NUM_IMAGES,THIS_IMAGE,UCOBOUND)
  external ATOMIC_DEFINE,ATOMIC_REF,CO_SUM,CO_MIN,CO_MAX
  external IMAGE_INDEX,LCOBOUND,NUM_IMAGES,THIS_IMAGE,UCOBOUND
end
subroutine ss
   entry ATOMIC_DEFINE
   entry ATOMIC_REF
   entry CO_SUM
   entry CO_MIN
   entry CO_MAX
   entry IMAGE_INDEX
   entry LCOBOUND
   entry NUM_IMAGES
   entry THIS_IMAGE
   entry UCOBOUND
end

File: C/0154/0154_0016.c
#include <stdio.h>

void sub(float a[], float b[], float n) {
  int i;
  b[0] = 0.0f;
  for (i = 1; i < n; ++i) {
    b[i] = a[i-1] * a[i]; 
  }
}
int main() {
  int i;
  float a[5000];
  float b[5000];

  for (i = 0; i < 5000; ++i) {
    a[i] = (float)i + 11.11f;
  }
  
  sub(a,b,5000);

  if (b[0]       == 0.000000f
      && b[9]    == 384.302124f
      && b[99]   == 12014.10253f
      && b[999]  == 1019312.062f
      && b[1999] == 4038532.000f
      && b[2999] == 9057753.000f 
      && b[3999] == 16076973.00f
      && b[4999] == 25096190.00f) {
    puts("OK");
  } else {
    puts("NG");
    printf("%f, %f, %f, %f, %f, %f, %f, %f\n",
	   b[0], b[9], b[99], b[999], b[1999], b[2999], b[3999], b[4999]);
  }
}
